pkg/cloud/azurepricesheet/downloader.go

```diff
@@ -0,0 +1,252 @@
+package azurepricesheet
+
+import (
+	"bufio"
+	"context"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/2020-09-01/commerce/mgmt/commerce"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+
+	"github.com/opencost/opencost/pkg/log"
+)
+
+type Downloader[T any] struct {
```

Comment:
I made this generic over the specific pricing type to avoid a circular dependency on `pkg/cloud.AzurePricing` in this code, but that might be too clever. Is there a sensible location I could move it to so it can be imported from `pkg/cloud` and `pkg/cloud/azurepricesheet`?

Reply:
I guess I could just move `AzurePricing` into `pkg/cloud/azurepricesheet`

Reply:
@babbageclunk what if you moved all of this + the existing azureProvider into a new package pkg/cloud/azure? You have made the decision (that I concur with) to introduce a new azure specific subpackage (azurepricesheet) - it could be refactored to just azure and include the azureprovider.go to become a one stop shop for all azure specific logic. 

This would almost certainly break our enterprise builds downstream but I am happy to prepare a companion PR and take care of things here on our end. 

That seems more maintainable than a pseudo-generic - what do you think?

Reply:
That sounds like a great idea! @r2k1 had mentioned that there was a reorganisation happening in parallel to break the providers into separate packages. I'll make a quick PR to do that and remove the type parameter.